marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/BuildConstraints.hs

```diff
@@ -314,7 +314,7 @@ buildApplyInputsConstraintsV1 systemStart eraHistory marloweOutput tipSlot metad
     Nothing -> pure case nextMarloweTimeout contract of
       Nothing -> maxSafeSlot
       Just nextTimeout -> case utcTimeToSlotNo' nextTimeout of
-        Right slot -> slot
+        Right slot -> if slot > invalidBefore' then slot else maxSafeSlot
```

Comment:
This isn't correct - we need to recursively find the next timeout in the continuation, not default to the max safe slot. I think the handling for this needs to be in `nextMarloweTimeout`, not here (we need to pass `invalidBefore'` to `nextMarloweTimeout`)

Reply:
Thanks, I'll correct this and add a unit test targeting it.

Reply:
you can find a verified calculation for a safe time interval (from the Marlowe perspective) in here: https://github.com/input-output-hk/marlowe/pull/140/files#diff-5103266280c0b8c38d09f6ed611b66de8bb9dc6c197fd67fbe0d9a2e18053427R695

Reply:
Actually, I don't think we can use this reference code because of the slot <-> milliseconds roundoff and the fact that Marlowe uses closed intervals but Cardano uses half-open intervals.